src/native/s3_client.c

```diff
@@ -687,6 +681,46 @@ static int s_marshal_http_headers_to_buf(const struct aws_http_headers *headers,
     return AWS_OP_SUCCESS;
 }
 
+static jobject s_get_error_type_enum(JNIEnv *env, int error_code) {
```

Comment:
I'll vote for a generic API in the exception class. And replace the `isErrorRetryable` in HTTP as well